pkg/api/pod/warnings.go

```diff
@@ -237,7 +236,7 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 		// use of container AppArmor annotation without accompanying field
 
 		isPodTemplate := fieldPath != nil // Pod warnings are emitted through applyAppArmorVersionSkew instead.
-		hasAppArmorField := hasPodAppArmorProfile || (c.SecurityContext != nil && c.SecurityContext.AppArmorProfile != nil)
```

Comment:
Note to reviewers: we weren't checking that the container annotation matched the pod profile here. Rather than extracting the logic from the pod strategy that does that comparison, I decided to just always emit a warning on pod templates when there is a pod-level field but a container level annotation.